### PR TITLE
Email testing

### PR DESCRIPTION
**Pull request by narekhovhannisyan**
## Motivation
Implement email testing API and `Inboxes`, `Projects`, `Messages` resources.

## Changes

- Separated axios error handler.
- Initialized Testing, Inboxes, Messages and Projects resources.
- Added usage samples.

### CODE REVIEW

File: src/lib/api/Testing.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  public messages: MessagesApi;
+
+  constructor(client: AxiosInstance, testInboxId?: number, accountId?: number) {
+    this.client = client;
+    if (!testInboxId) {
+      // eslint-disable-next-line no-console
+      console.warn(TEST_INBOX_ID_MISSING);
+    }
+    this.testInboxId = testInboxId;
+    if (!accountId) {
```
Comment (VladimirTaytor) [inline src/lib/api/Testing.ts:38]: As I'm looking at this right now, we will always show this warning for users even if they don't want to use Testing Client
- #idea Maybe let's add in `src/lib/mailtrap-client.ts` lazy initialisation for testing client?
- wdyt?

**Review by VladimirTaytor — COMMENTED** [review]
Looks good in general left few suggestions 👌

File: src/lib/api/Testing.ts (hunk 1 above)
Comment (narekhovhannisyan) [reply src/lib/api/Testing.ts:38]: lazy loading will bring the following syntax:

```ts
const client = new MailtrapClient({ token: TOKEN, testInboxId: TEST_INBOX_ID, accountId: ACCOUNT_ID });
const testingClient = client.testing()

testingClient.projects.get()
```

Comment (narekhovhannisyan) [reply src/lib/api/Testing.ts:38]: @VladimirTaytor I think we can introduce a getter method for testing and while users will try to access testing, they will receive warning messages. This way we don't have to call testing. Wdyt?

File: examples/testing/inboxes.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const projectsClient = client.testing.projects
+const inboxesClient = client.testing.inboxes
+
+projectsClient.getList()
+  .then(async (projects) => {
+    if (projects && projects.length > 0) {
+      const firstProjectId = projects[0].id
+
+      // const inbox = await inboxesClient.create(firstProjectId, 'test-inbox')
```
Comment (VladimirTaytor) [inline examples/testing/inboxes.ts:17]: If we want to keep this commented out we need some description

**Review by VladimirTaytor — APPROVED** [review]
Looks good, would be nice to test everything manually before delivering a new release

File: src/lib/api/resources/Inboxes.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+   */
+  public async resetCredentials(inboxId: number) {
+    const url = `${this.inboxesURL}/${inboxId}/reset_credentials`;
+
+    try {
+      const apiRespone = await this.client.patch<Inbox>(url);
+
+      return apiRespone.data;
+    } catch (error) {
+      return handleSendingError(error);
```
Comment (mklocek) [inline src/lib/api/resources/Inboxes.ts:133]: I wonder if `handleSendingError` could be done once in axios response interceptor to save us some code duplication. Not a must though

File: src/lib/axios-logger.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+/**
+ * Error handler for axios response.
+ */
+export default function handleSendingError(error: AxiosError | unknown) {
+  if (axios.isAxiosError(error)) {
+    const serverErrorsObject =
+      error.response?.data &&
+      typeof error.response.data === "object" &&
```
Comment (mklocek) [inline src/lib/axios-logger.ts:24]: 👍 

Great that you're handling it. It looks that our API docs don't mention it, but we have 422 error responses which usually return an object.

**Review by mklocek — APPROVED** [review]

**Review by leonid-shevtsov — APPROVED** [review]

File: src/lib/api/resources/Inboxes.ts (hunk 3 above)
Comment (narekhovhannisyan) [reply src/lib/api/resources/Inboxes.ts:133]: Agree, forgot about it :) checking

**Pull request merged**